functions/api/category.js

```diff
@@ -0,0 +1,386 @@
+    logger.info(`Creating category in firestore....`);
+    // Validate parameters
+    logger.debug('Validating params.')
+    const { error } = validateParams(request.body, constants.CATEGORY)
+    if (error) {
+        const err = new Error(error.details[0].message)
+        err.statusCode = 400
+        next(err)
+        return;
+    }
+
+    // If category already exists, return 400
+    var categoryName = request.body.category.toLocaleLowerCase()
+    logger.info(`Creating category ${categoryName} in firestore....`);
+    const doc = db.collection(constants.CATEGORY).doc(categoryName);
+    const category = await doc.get()
+    if (category.exists) {
+        const err = new Error(`The category ${categoryName} already exists. Please update if needed.`)
+        err.statusCode = 400
+        next(err)
+        return;
+    }
+
+    let data = {}
+    data[constants.DESCRIPTION] = request.body.description
+    data[constants.IS_ACTIVE] = true
+    data[constants.CREATED_DATE] = new Date()
+    data[constants.LAST_UPDATED_DATE] = new Date()
+    data[constants.PRODUCTS] = []
+    await db.collection(constants.CATEGORY).doc(categoryName).set(data)
+    logger.debug(`${categoryName} document Created`)
+    var result = {}
+    result[constants.CATEGORY] = categoryName
+    result[constants.DESCRIPTION] = data.description
+    response.status(200).send(result);    
+});
+
+/**
+ * @description Route to update description of category
+ * @returns  updated category
+ * @throws 400 if category does not exist or has wrong params
+ */
+router.put('/desc', async (request, response, next) => {
+    logger.info(`Updating description for category in firestore....`);
+    
+    // Validate parameters
+    const { error } = validateParams(request.body, constants.DESCRIPTION)
+    if (error) {
+        const err = new Error(error.details[0].message)
+        err.statusCode = 400
+        next(err)
+        return;
+    }
+
+    // If category does not exists, return 400
+    var categoryName = request.body.category
+    logger.info(`Updating description for category ${categoryName} in firestore....`);
+    const categoryRef = db.collection(constants.CATEGORY).doc(categoryName);
+    const category = await categoryRef.get()
+    if (!category.exists) {
+        const err = new Error(`Requested category ${categoryName} is not present in Firestore.`)
+        err.statusCode = 404
+        next(err)
+        return;
+    }
+    let data = {}
+    data[constants.DESCRIPTION] = request.body.description
+    data[constants.LAST_UPDATED_DATE] = new Date()
+    await categoryRef.update(data)
+    delete data[constants.LAST_UPDATED_DATE]
+    data[constants.CATEGORY] = categoryName
+    data = JSON.parse(JSON.stringify( data, [constants.CATEGORY,constants.DESCRIPTION]));
+    logger.debug(`Updated description of category ${categoryName} to ${request.body.description}`)
+    response
+        .status(200)
+        .send(data);
```

Comment:
do we need to send a data back? any reasons? can we have it simple - 204 no content.

Reply:
No reason as such. But UI guys didn't said anything after I sent them the contract.

Reply:
Could we check with them, if they need. Because from backend we create an object , pick data from here and there and send them. Both in POST and PUT, UI will have access to user's input. So when they get a 200/204 response from backend, they can update.